lib/fluent/plugin/out_influxdb.rb

```diff
@@ -105,14 +107,14 @@ def write(chunk)
     points = []
     chunk.msgpack_each do |tag, time, record|
       timestamp = record.delete(@time_key) || time
-      if tag_keys.empty?
+      if tag_keys.empty? && !@auto_tags
         values = record
         tags = {}
       else
         values = {}
         tags = {}
         record.each_pair do |k, v|
-          if @tag_keys.include?(k)
+          if @auto_tags and v.is_a? String or @tag_keys.include?(k)
```

Comment:
Sorry for the late response. Mixing `and` and `&&` is bad practice.
Use `(@auto_tags && v.is_a?(String)) || @tag_keys.include?(k)` instead.